app/_src/guides/logging.md

```diff
@@ -0,0 +1,181 @@
+```
+**Option 2: Port forwarding** 
+The Envoy interface can also be accessed directly. This can be achieved using Kubernetes port-forwarding capabilities, as the Envoy admin interface is not exposed by default.
+
+```shell
+kubectl port-forward kuma-demo-app-76df4d8cf5-9q9j9 -n kuma-demo 9901:9901 2>&1 &
+```
+
+Once port forwarding is set up, you can increase the log level by sending an `HTTP POST` request with one of the supported values. The debug-level logs will then be available via the standard logging facilities (for example, stderr on Kubernetes):
+
+```shell
+curl -X POST http://localhost:9901/logging?level=debug
+```
+Or targeting a particular component:
+
+```shell
+curl -X POST http://localhost:9901/logging?wasm=debug
+```
+
+{% endtab %}
+{% tab Universal %}
+
+In Universal mode, logging can be enabled by passing the `--envoy-log-level` flag to the kuma-dp process.
+
+```shell
+kuma-dp run --envoy-log-level=debug
+```
+
+You can optionally target specific components, each with its own log level, by passing the flag:
+`--envoy-component-log-level`
+
+```shell
+kuma-dp run --envoy-component-log-level="upstream:debug,connection:trace"
+```
+
+{% endtab %}
+{% endtabs %}
+
+
+## Adjusting logging levels for Kuma-CP
+The available logging levels for Control Planes are:
+
+* debug
+* info
+* off
+
+
+{% tabs %}
+{% tab Kubernetes %}
+When using Helm to deploy on Kubernetes, the following can be used to change the Control Plane log level:
+
+```yaml
+kuma:
+  controlPlane:
+   logLevel: "debug"
+```
+
+Additionally, the --log-level flag can be passed as an argument to the kuma-cp binary in your container's definition.
+
+```yaml
+containers:
+  - name: control-plane
+    image: "docker.io/kong/kuma-cp:2.9.3"
+    args:
+      - run
+      - --log-level=debug
+      - --config-file=/etc/kuma.io/kuma-control-plane/config.yaml
+```
+
+{% endtab %}
+{% tab Universal %}
+
+In Universal mode, logging can be enabled by passing the `--envoy-log-level` flag to the kuma-dp process.
+
+```shell
+kuma-cp run --log-level=debug
+``` 
+
+{% endtab %}
+{% endtabs %}
+
+## Adjusting logging levels for CoreDNS
+Logging for CoreDNS does not have specific levels; it is either on/true or off/false.
+
+
+{% tabs %}
+{% tab  Kubernetes %}
+When using Helm to deploy on Kubernetes, the following can be used to change the DNS log level:
+
+```yaml
+kuma:
+  dataPlane:
+    dnsLogging: true
+```
+
+Additionally, the environment variable `KUMA_RUNTIME_KUBERNETES_INJECTOR_BUILTIN_DNS_LOGGING` can be set in your container's definition.
+
+```yaml
+      containers:
+        - name: control-plane
+          image: "docker.io/kong/kuma-cp:2.9.3"
+          env:
+            - name: KUMA_RUNTIME_KUBERNETES_INJECTOR_BUILTIN_DNS_LOGGING
+              value: "true"
+```
+
+{% endtab %}
+{% tab Universal %}
+
+In Universal mode, logging can be enabled by setting the environment variable `KUMA_RUNTIME_KUBERNETES_INJECTOR_BUILTIN_DNS_LOGGING`.
```

Comment:
Universal DPs use `KUMA_DNS_ENABLE_LOGGING`, insead of `KUMA_RUNTIME_KUBERNETES_INJECTOR_BUILTIN_DNS_LOGGING`